core/src/main/resources/lib/hudson/newFromList/form.jelly

```diff
@@ -49,26 +49,21 @@ THE SOFTWARE.
     </st:attribute>
   </st:documentation>
 
-
+  <!-- hide form initially to avoid jerky display (class add-item set by add-item.js) -->
+  <style>
+    #main-panel form {min-height:40em; background:#fcfcfc; max-width: 60em; border-radius:3px; border:1px solid #eee;}
+    form table{display:none;} 
+    body.add-item #main-panel form{background:none; min-height:0; border:none;}
+    body.add-item form table{display:table;} 
+  </style>
+  
   <j:set var="descriptors" value="${h.filterDescriptors(it,attrs.descriptors)}" />
-
   <s:form method="post" action="${attrs.action?:'createItem'}" name="createItem">
-    <s:entry title="${attrs.nameTitle}">
-      <s:textbox id="name" name="name" checkUrl="'${h.jsStringEscape(attrs.checkUrl)}?value='+encodeURIComponent(this.value)"
-                 onchange="updateOk(this.form)" onkeyup="updateOk(this.form)" />
-      <script>$('name').focus();</script>
-    </s:entry>
+    <div id="j-add-item-form-data" data-root="${rootURL}/"></div>
+
+    <l:js src="jsbundles/add-item.js" />
+    <l:css src="jsbundles/jenkins-widgets.css" />  
 
-    <j:forEach var="descriptor" items="${descriptors}">
```

Comment:
:bug: whatever you are doing here cannot be right, since `descriptors` is being set, but not then used. You are aware that this tag is used by things other than `newJob.jelly` and having nothing to do with `TopLevelItemDescriptor`, right? Like creating new agents, etc.? Perhaps you meant to leave this tag untouched, and just edit `newJob.jelly` to use the replacement API.